### ISSUE TITLE
org-ref-insert-ref-link does not show all equations

### ISSUE THREAD

**Issue by roiholtzman** [jkitchin/org-ref#801]
When in an org document and executing `org-ref-insert-ref-link` a `helm` buffer opens with only few of the equations that are in the document.
Why is that?

**Comment by jkitchin** [comment]
I don't know. Can you provide a small example that has labels that don't show?

**Comment by roiholtzman** [comment]
Sure. I wrote this short document:

```
Example for how equation labels are not shown when inserting a reference.

Equation one is
\begin{equation}
\label{eq:1}
A = B
\end{equation}
And equation two is
\begin{equation}
\label{eq:2}
C = D.
\end{equation}

A reference is Eq.  
```

When at the end of the file I insert a reference with `, i r` I get both options: `eq:1, eq:2` which is good.
<img width="1552" alt="Screen Shot 2020-10-04 at 16 39 09" src="https://user-images.githubusercontent.com/7188052/95017177-72591580-0660-11eb-95e4-f96929b812e2.png">


But if I change the label of eq:2 to say eq:3, and try to reference it with `, i r` then I only get `eq:3` in the option list.
<img width="1552" alt="Screen Shot 2020-10-04 at 16 37 50" src="https://user-images.githubusercontent.com/7188052/95017182-7d13aa80-0660-11eb-8806-3eb540276bb9.png">

**Comment by jkitchin** [comment]
I can confirm this happens.  I am not real sure why yet. The labels are updated via font lock, and this seems to not work on this example for some reason. 

One temporary fix  may be to disable font-lock use for this like this:

`(setq org-ref-label-use-font-lock nil)`

For big documents this may be slow. I don't have time now to figure out what is happening with the font-lock issue right now. The issue is happening in the functions `org-ref-add-labels` and `org-ref-delete-labels`. These are hard to debug though.

**Comment by fleimgruber** [comment]
I can't reproduce this using a recent (few days) Spacemacs (which is what OP seems to use, @roiholtzman please confirm) and org-ref-20201231.1406 from MELPA. `setq org-ref-label-use-font-lock` is set to `t`.

**Comment by jkitchin** [comment]
I think I have finally fixed a lot of the issue here in 1792166. 

It should work reliably with `(setq org-ref-label-use-font-lock nil)`. With that setting it will search the whole buffer each time, but that is a regex based search, so it should be fast. I think it should work now for `(setq org-ref-label-use-font-lock t)` also.
